### PR TITLE
Refractor: remove `processFindOptions` and `getPermissionQuery` methods declaration in repositoryClient

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/server/saved_objects/types.ts [hunk 1]
```diff
@@ -111,7 +112,15 @@ export interface SavedObjectsFindOptions {
...
   workspaces?: string[];
-  queryDSL?: Record<string, any>;
+  /**
+   * The params here will be combined with bool clause and is used for filtering with ACL structure.
+   */
+  ACLSearchParams?: {
+    workspaces?: string[];
+    principals?: Principals;
+    workspacePermissionModes?: string[];
+    objectPermissionModes?: string[];
```
Comment (wanglam) [inline src/core/server/saved_objects/types.ts:122]: Could we merge `workspacePermissionModes` and `objectPermissionModes` into one permission modes object?

**Review by xluo-aws — APPROVED** [review]

File: src/core/server/saved_objects/types.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/types.ts:122]: As now the permissionModes here is a "or" combination, I think we can merge these two modes into one.

Comment (wanglam) [reply src/core/server/saved_objects/types.ts:122]: Is that means we can't merge these two modes into one?

Comment (SuZhou-Joe) [reply src/core/server/saved_objects/types.ts:122]: We can, will do that

File: src/core/server/saved_objects/service/lib/search_dsl/query_params.ts [hunk 2]
```diff
@@ -286,9 +286,9 @@ export function getQueryParams({
 
   if (ACLSearchParams) {
     const shouldClause: any = [];
-    if (ACLSearchParams.workspacePermissionModes && ACLSearchParams.principals) {
+    if (ACLSearchParams.permissionModes && ACLSearchParams.principals) {
       const workspacePermissionDSL = ACL.generateGetPermittedSavedObjectsQueryDSL(
```
Comment (wanglam) [inline src/core/server/saved_objects/service/lib/search_dsl/query_params.ts:290]: How about rename the `workspacePermissionDSL` to `permissionDSL`?

File: src/core/server/saved_objects/types.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/types.ts]: Done.

File: src/core/server/saved_objects/service/lib/search_dsl/query_params.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/service/lib/search_dsl/query_params.ts:290]: Sure, good catch

**Review by wanglam — APPROVED** [review]
LGTM

**Pull request merged**